tests/test_installer.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2024 Science project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import subprocess
+
+from _pytest.tmpdir import TempPathFactory
+
+INSTALLER = "./install.sh"
+
+
+def test_installer_help():
+    """Validates -h|--help in the installer."""
+    assert "--help" in subprocess.check_output([INSTALLER, "-h"]).decode("utf-8")
+    assert "--help" in subprocess.check_output([INSTALLER, "--help"]).decode("utf-8")
+
+
+def test_installer_fetch_latest(tmp_path_factory: TempPathFactory):
+    """Invokes install.sh to fetch the latest science release binary, then invokes it."""
+    test_dir = tmp_path_factory.mktemp("install-test-default")
+    assert "success" in subprocess.check_output(
```

Comment:
This test fails on the macos-12 (x86-64) shard as does the one below in the same check. Since the output is swallowed in both cases there is not much to go on currently. You'll have to experiment.

Reply:
appears from this debug log output (which also catches an error-mode print bug) to be related to output of `arch` vs `uname`, : https://github.com/a-scie/lift/actions/runs/10721551555/job/29749027884?pr=88#step:9:256

on my x86 mac, `arch` says `i386` and `uname -m` says `x86_64`.